src/Webapi/Webapi__Dom/Webapi__Dom__Element.re

```diff
@@ -5,18 +5,13 @@ let ofNode = (node: Dom.node) : option('a) =>
     None;
 
 module Impl = (T: {type t;}) => {
-  let asHtmlElement: T.t => Js.null(Dom.htmlElement) = [%bs.raw
-    {|
-    function (element) {
-      // BEWARE: Assumes "contentEditable" uniquely identifies an HTMLELement
-      return element.contentEditable !== undefined ?  element : null;
+  let asHtmlElement: T.t => option(Dom.htmlElement) = [%raw {|
+    function(element) {
+      if (element instanceof HTMLElement) return element;
     }
-  |}
-  ];
-  [@deprecated "asHtmlElement uses a weak heuristic, consider using unsafeAsHtmlElement instead"]
-  let asHtmlElement: T.t => option(Dom.htmlElement) =
-    (self) => Js.Null.toOption(asHtmlElement(self));
+  |}];
 
+  [@deprecated "Unsafe cast, use [asHtmlElement] instead"]
```

Comment:
This deprecation warning is a bit onerous for library authors building on top of `bs-webapi`, I'm going to have to add a non-deprecated replacement and further diverge from using `bs-webapi` directly.

I have 11 places in my code where I know with certainty that I'm in a HTML tree structure, but instead of lightweight code:
`switch(elm->Element.parentElement->Element.unsafeAsHtmlElement)`
which generates
```
    var parent = elm$1.parentElement;
    if (parent == null) {
      // no parent code
    }
```
I now have to do:
`switch(elm->HtmlElement.parentElement->Belt.Option.flatMap(Element.asHtmlElement))`
which generates 
```
    var parent = Belt_Option.flatMap(Caml_option.nullable_to_opt(elm.parentElement), Webapi__Dom__Element.asHtmlElement);
    if (parent === undefined) {
      // no parent code
    }
```
(it could be shrunk with nested switch statements instead of flatMap, but that's unacceptable imo)

This difference is partly due to how `nullable` externals are generated. When used immediately they are handled with optimal JS. When passed as values, however, they have to be wrapped in a `nullable_to_opt` function call to normalise them with regular option values.

Reply:
Yeah, you're right. The function names are already marked as `unsafe`. I'll publish a new version and un-deprecate them.

Reply:
Thanks!